contracts/modules/Offers/V1/OffersV1.sol

```diff
@@ -0,0 +1,422 @@
+    //     |                  |----.
+    //     |                  |    | delete offer from contract
+    //     |                  |<---'
+    //   Caller           ,---+----.                ,---------+----------.
+    //     ,-.            |OffersV1|                |ERC721TransferHelper|
+    //     `-'            `--------'                `--------------------'
+    //     /|\
+    //      |
+    //     / \
+    /// @notice Fills a given offer for an owned NFT, in exchange for ETH/ERC-20 tokens
+    /// @param _tokenContract The address of the ERC-721 token to transfer
+    /// @param _tokenId The ID of the ERC-721 token to transfer
+    /// @param _offerId The ID of the offer to fill
+    /// @param _currency The address of the ERC-20 to take, or address(0) for ETH
+    /// @param _amount The amount to take
+    /// @param _finder The address of the offer referrer
+    function fillOffer(
+        address _tokenContract,
+        uint256 _tokenId,
+        uint256 _offerId,
+        address _currency,
+        uint256 _amount,
+        address _finder
+    ) external nonReentrant {
+        Offer storage offer = offers[_tokenContract][_tokenId][_offerId];
+
+        require(offer.maker != address(0), "fillOffer must be active offer");
+        require(IERC721(_tokenContract).ownerOf(_tokenId) == msg.sender, "fillOffer must be token owner");
+        require(offer.currency == _currency && offer.amount == _amount, "fillOffer _currency & _amount must match offer");
+
+        // Payout respective parties, ensuring royalties are honored
+        (uint256 remainingProfit, ) = _handleRoyaltyPayout(_tokenContract, _tokenId, offer.amount, offer.currency, USE_ALL_GAS_FLAG);
+
+        // Payout optional protocol fee
+        remainingProfit = _handleProtocolFeePayout(remainingProfit, offer.currency);
+
+        // Payout optional finders fee
+        if (_finder != address(0)) {
+            uint256 findersFee = (remainingProfit * offer.findersFeeBps) / 10000;
+            _handleOutgoingTransfer(_finder, findersFee, offer.currency, USE_ALL_GAS_FLAG);
+
+            remainingProfit -= findersFee;
+        }
+
+        // Transfer remaining ETH/ERC-20 tokens to offer taker
+        _handleOutgoingTransfer(msg.sender, remainingProfit, offer.currency, USE_ALL_GAS_FLAG);
+
+        // Transfer NFT to offer maker
+        erc721TransferHelper.transferFrom(_tokenContract, msg.sender, offer.maker, _tokenId);
+
+        ExchangeDetails memory userAExchangeDetails = ExchangeDetails({tokenContract: offer.currency, tokenId: 0, amount: offer.amount});
+        ExchangeDetails memory userBExchangeDetails = ExchangeDetails({tokenContract: _tokenContract, tokenId: _tokenId, amount: 1});
+
+        emit ExchangeExecuted(offer.maker, msg.sender, userAExchangeDetails, userBExchangeDetails);
+        emit OfferFilled(_tokenContract, _tokenId, _offerId, msg.sender, _finder, offer);
+
+        delete offers[_tokenContract][_tokenId][_offerId];
```

Comment:
Asking to learn: from what I understand, this deletes [the metadata for a given offer](https://github.com/ourzora/v3/pull/126/files#diff-e217b3adfc705cc4b92f06b296679588ff1730ad3fa7a118f3c0c0a4238b1cf0R42-R44). However, there is no explicit call to interact with the [`offersForNft` mapping](https://github.com/ourzora/v3/pull/126/files#diff-e217b3adfc705cc4b92f06b296679588ff1730ad3fa7a118f3c0c0a4238b1cf0R46-R48) afterwards. Is this design intended to maintain the history of offers for the NFT, even for offers that have already been fulfilled?

Reply:
haha originally we designed to delete offer IDs in addition to the metadata after fill/cancel, but ironically this ends up raising gas costs as we'd need to keep track of each offer ID's index in its associated `offersForNFT` array. so the tradeoff we settled for was keeping the history of every offer, as you mentioned, but having costs be relatively lower

Reply:
@kulkarohan - Thank you for that context around tradeoffs; I learned something new. 👍 

I'll go ahead and resolve this comment from the PR conversation. ✌️